File: swe2/mod2/passwordHashing/java/src/main/java/org/example/Hash.java
package org.example;

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

public class Hash 
{
    public static void main( String[] args ) {

        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
        String hashedPassword = passwordEncoder.encode("your password");

        System.out.println(hashedPassword );

        boolean isMatch = passwordEncoder.matches("your password", "$2b$10$AQXoVkfzAovJ9RHTtmd6N.Yegy3V9ALTlYDcCM76HxBqq044q6xLK");
    }
}

File: swe2/mod2/passwordHashing/js/hash.js
const bcrypt = require('bcrypt')

// hash the password 'password101'
bcrypt.hash('password101', 10).then(console.log)

// compare a password with a hash
bcrypt.compare('password101', '$2b$10$AQXoVkfzAovJ9RHTtmd6N.Yegy3V9ALTlYDcCM76HxBqq044q6xLK').then(console.log)